format-config-v1_test.go

```diff
@@ -0,0 +1,191 @@
+	for index := range jbod {
+		jbod[index] = getUUID()
+	}
+	for index := range jbod {
+		formatConfigs[index] = &formatConfigV1{
+			Version: "1",
+			Format:  "xl",
+			XL: &xlFormat{
+				Version: "1",
+				Disk:    jbod[index],
+				JBOD:    jbod,
+			},
+		}
+	}
+	for index := range jbod {
+		jbod[index] = getUUID()
+	}
+	// Corrupt JBOD entries on disk 6 and disk 8.
+	formatConfigs[5].XL.JBOD = jbod
+	formatConfigs[7].XL.JBOD = jbod
+	return formatConfigs
+}
+
+// generates a invalid format.json Disk UUID for XL backend.
+func genFormatXLInvalidDisks() []*formatConfigV1 {
+	jbod := make([]string, 8)
+	formatConfigs := make([]*formatConfigV1, 8)
+	for index := range jbod {
+		jbod[index] = getUUID()
+	}
+	for index := range jbod {
+		formatConfigs[index] = &formatConfigV1{
+			Version: "1",
+			Format:  "xl",
+			XL: &xlFormat{
+				Version: "1",
+				Disk:    jbod[index],
+				JBOD:    jbod,
+			},
+		}
+	}
+	// Make disk 5 and disk 8 have inconsistent disk uuid's.
+	formatConfigs[4].XL.Disk = getUUID()
+	formatConfigs[7].XL.Disk = getUUID()
+	return formatConfigs
+}
+
+// generates a invalid format.json Disk UUID in wrong order for XL backend.
+func genFormatXLInvalidDisksOrder() []*formatConfigV1 {
+	jbod := make([]string, 8)
+	formatConfigs := make([]*formatConfigV1, 8)
+	for index := range jbod {
+		jbod[index] = getUUID()
+	}
+	for index := range jbod {
+		formatConfigs[index] = &formatConfigV1{
+			Version: "1",
+			Format:  "xl",
+			XL: &xlFormat{
+				Version: "1",
+				Disk:    jbod[index],
+				JBOD:    jbod,
+			},
+		}
+	}
+	// Re order jbod for failure case.
+	jbod[1], jbod[2] = jbod[2], jbod[1]
+	formatConfigs[2].XL.JBOD = jbod
+	jbod[3], jbod[4] = jbod[1], jbod[3]
+	formatConfigs[4].XL.JBOD = jbod
+	return formatConfigs
+}
+
+// Wrapper for calling FormatXL tests - currently validates
+//  - valid format
+//  - unrecognized version number
+//  - invalid JBOD
+//  - invalid Disk uuid
+func TestFormatXL(t *testing.T) {
+	formatInputCases := [][]*formatConfigV1{
+		genFormatXLValid(),
+		genFormatXLInvalidVersion(),
+		genFormatXLInvalidJBOD(),
+		genFormatXLInvalidDisks(),
+		genFormatXLInvalidDisksOrder(),
+	}
+	testCases := []struct {
+		formatConfigs []*formatConfigV1
+		shouldPass    bool
+	}{
+		{
+			formatConfigs: formatInputCases[0],
+			shouldPass:    true,
+		},
+		{
+			formatConfigs: formatInputCases[1],
+			shouldPass:    false,
+		},
+		{
+			formatConfigs: formatInputCases[2],
+			shouldPass:    false,
+		},
+		{
+			formatConfigs: formatInputCases[3],
+			shouldPass:    false,
+		},
+		{
+			formatConfigs: formatInputCases[4],
+			shouldPass:    false,
+		},
+	}
+
+	for i, testCase := range testCases {
+		err := checkFormatXL(testCase.formatConfigs)
+		if err != nil && testCase.shouldPass {
+			t.Errorf("Test %d: Expected to pass but failed with %s", i+1, err)
+		}
+		if err == nil && !testCase.shouldPass {
+			t.Errorf("Test %d: Expected to fail but passed instead", i+1)
+		}
+	}
+}
```

Comment:
Instead of this style, it is better to write 1-1 unit tests. For example, isSavedUUIDInOrder will have isSavedUUIDInOrderTest. This test will only look out for variations in UUID order. This way, we will have a cleaner coverage for all functions. If we only test higher level functions, details may be lost. It is hard to remember all the tricky corner cases at higher level functions. Test functions for higher level functions will only test the glue code, how lower level functions are stitched together.